### PR TITLE
Add IoT Policy Allows Action as * query for CloudFormation #1109

### PR DESCRIPTION
**Pull request by pedro-mimoso**
IoT Policy should not allow Action to be set as * #1109

### CODE REVIEW

File: assets/queries/cloudFormation/iot_policy_allows_action/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "iot_policy_allows_action",
+  "queryName": "IoT Policy Allows Action as *",
+  "severity": "MEDIUM",
+  "category": null,
+  "descriptionText": "IoT Policy should not allow Action to be set as *",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iot-policy.html"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/iot_policy_allows_action/query.rego [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+	result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.Properties.PolicyDocument.Statement.Effect=%s.Action",[name,statements[k].Effect]),   
+                "issueType":		"IncorrectValue",  
+                "keyExpectedValue": sprintf("Resources.%s.Properties.PolicyDocument.Statement[%s].Action is not '*'",[name,statements[k].Effect]),
+                "keyActualValue": 	sprintf("Resources.%s.Properties.PolicyDocument.Statement[%s].Action is '*'",[name,statements[k].Effect]),
+              }
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/query.rego:18]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/iot_policy_allows_action/test/negative.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      Properties:
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - iot:Connect
+            Resource:
+            - arn:aws:iot:us-east-1:123456789012:client/client1
+        PolicyName: PolicyName
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/test/negative.yaml:15]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/iot_policy_allows_action/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+          - Effect: Allow
+            Action: "*"
+            Resource:
+            - arn:aws:iot:us-east-1:123456789012:client/client
+          - Effect: Deny
+            Action:
+            - sqs:*
+            NotResource:
+            - !GetAtt myqueue.Arn
+        PolicyName: PolicyName
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/test/positive.yaml:19]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/iot_policy_allows_action/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+          Statement:
+          - Effect: Allow
+            Action: "*"
+            Resource:
+            - arn:aws:iot:us-east-1:123456789012:client/client
+          - Effect: Deny
+            Action:
+            - sqs:*
+            NotResource:
+            - !GetAtt myqueue.Arn
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/test/positive.yaml:18]: Resource `myqueue` does not exist in this template. Remove the `GetAtt `function and leave the ARN hardcoded:

`NotResource: my-hardcoded-arn`

File: assets/queries/cloudFormation/iot_policy_allows_action/metadata.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "iot_policy_allows_action",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/metadata.json:2]: Change **directory name** to: `iot_policy_allows_wildcard_action`
Change **id** to: `iot_policy_allows_wildcard_action`
Change **query name** to: `IoT Policy Allows Action as Wildcard`

File: assets/queries/cloudFormation/iot_policy_allows_action/metadata.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  "queryName": "IoT Policy Allows Action as *",
+  "severity": "MEDIUM",
+  "category": null,
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/iot_policy_allows_action/metadata.json:5]: `Identity and Access Management`

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Almost good

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**